test-d/simplify.ts

```diff
@@ -42,3 +42,25 @@ expectType<SomeInterfaceAsTypeWrittenByHand>(valueAsInterface);
 expectAssignable<Record<string, unknown>>(valueAsLiteral);
 expectAssignable<Record<string, unknown>>(valueAsSimplifiedInterface);
 expectNotAssignable<Record<string, unknown>>(valueAsInterface); // Index signature is missing in interface
+
+// Should return the original type if it is not simplifiable, like a function.
+type SomeFunction = (type: string) => string;
+expectType<Simplify<SomeFunction>>((type: string) => type);
+
+// Sould simplify to public interface (what it is currently doing)? Return the Class type? Or something else?
```

Comment:
I also wonder about classes. Currently it returns a simplified public version of the class interface.
What do you expect? the public interface? complete interface? or just the original type?

Reply:
I would say, just the original type. The point of the utility is to simplify nested and complicated types.